lib/core.ps1

```diff
@@ -1005,9 +1005,14 @@ function shim($path, $global, $name, $arg) {
         warn_on_overwrite "$shim.cmd" $path
         @(
             "@rem $resolved_path",
-            "@bash `"`$(wslpath -u '$resolved_path')`" $arg %* 2>nul",
-            '@if %errorlevel% neq 0 (',
-            "  @bash `"`$(cygpath -u '$resolved_path')`" $arg %* 2>nul",
+            '@bash -c "command -v wslpath >/dev/null 2>&1"',
+            '@if %errorlevel% equ 0 (',
+            "  @bash `"`$(wslpath -u '$resolved_path')`" $arg %*",
+            ') else (',
+            '  @bash -c "command -v cygpath >/dev/null 2>&1"',
+            '  @if %errorlevel% equ 0 (',
+            "    @bash `"`$(cygpath -u '$resolved_path')`" $arg %*",
+            '  )',
```

Comment:
```suggestion
            '@echo off',
            'bash -c "command -v wslpath >/dev/null"',
            'if %errorlevel% equ 0 (',
            "  bash `"`$(wslpath -u '$resolved_path')`" $arg %*",
            ') else (',
            "  bash `"`$(cygpath -u '$resolved_path')`" $arg %*",
```

The command `command -v wslpath` outputs the filepath if it is found, or `null` if not. Therefore, redirecting stdout should suffice.

Since all Linux environments under Windows should have `wslpath` (WSL) or `cygpath` (Cygwin, MSYS2), the second `if` check is unnecessary and could be omitted.

Reply:
Makes sense, thanks